src/ray/rpc/gcs_server/gcs_rpc_client.h

```diff
@@ -54,7 +54,7 @@ class Executor {
     auto operation_callback = [this, request, callback, executor](                     \
                                   const ray::Status &status,                           \
                                   const METHOD##Reply &reply) {                        \
-      if (!status.IsIOError()) {                                                       \
+      if (!status.IsIOError() && !status.IsGrpcUnavailable()) {                        \
```

Comment:
Hmm should IsIOError return True for unavailable? It seems like a type of IO error.

Reply:
Maybe gRPC unavailable should be a flag on IOError, like permanently down.

Reply:
I looked at the code a bit more, here actually what we want to retry is grpc error, it's just grpc error was represented as an IOError. For potential IOError returned by the ray code, it's captured inside GcsStatus as part of the Reply object. So now I use GrpcUnavailable and GrpcUnknown to capture all grpc errors and have a `IsGrpcError()` to check that.